enhancements/sig-architecture/11-cluster-deletion/README.md

```diff
@@ -0,0 +1,59 @@
+# Enhancement of cluster deletion 
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+Currently, when we remove a cluster from the Hub, the `managedClusterAddons` and `manifestWorks` in the cluster namespace need to be deleted manually, and cannot ensure that the resources deployed on the Spoke cluster can be cleaned up. This proposal is to define clearly what happened when a cluster is deleted.
+
+## Motivation
+
+### Goals
+
+- Define which resources related the deleted cluster should be cleaned up by OCM.
+- Clarify the order in which these resources are deleted.
+
+### Non-Goals
+
+- Clean up customized resources automatically.
+
+## Proposal
+
+### Which resources should be cleaned up when a cluster is removed from the Hub.
+
+On the Hub cluster
+
+1. The `managedCluster` should be deleted.
+2. The `managedClusterAddons` and `manifestWorks` in the cluster namespace should be cleaned up by OCM.
+3. The other resources in the cluster namespace except `managedClusterAddons` and `manifestWorks` should be cleaned up by their owners.
+4. The cluster namespace should be deleted too.
+   
+On the Spoke cluster
+
+1. The resources deployed by the `manifestWorks` should be cleaned up by OCM. 
+2. The resources deployed by other owners can be cleaned up optionally by their owners like the subscriptions deployed by application addon.
+3. The `Klusterlet` CR should be deleted.
+
+### The behaviors when a cluster is removed from the Hub.
+
+1. The first step to remove a cluster is to delete the `managedCluster` on the Hub.
+2. Delete the `managedClusterAddons` and `manifestWorks` in the cluster namespace.
```

Comment:
this needs more consideration. 

1. what should we do when cluster status is unknown. should we remove addon/work in this scenario?
2. should we define a grace period to remove these resources?

Reply:
I think we can force delete all managedClusterAddons and manifestWorks 10 minutes after the managedCluster is unavaliable.